plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/KtlintPlugin.kt

```diff
@@ -63,28 +68,37 @@ open class KtlintPlugin : Plugin<Project> {
         }
     }
 
-    private fun addKtLintTasksToAndroidKotlinPlugin(target: Project,
-                                                    extension: KtlintExtension) {
+    private fun addKtLintTasksToAndroidKotlinPlugin(target: Project, extension: KtlintExtension) {
         target.pluginManager.withPlugin("kotlin-android") {
             target.afterEvaluate {
                 val ktLintConfig = createConfiguration(target, extension)
 
-                target.extensions.findByType(BaseExtension::class.java)?.sourceSets?.forEach {
-                    val kotlinSourceDir = it.java.sourceFiles
-                    val runArgs = it.java.srcDirs.map { "${it.path}/**/*.kt" }.toMutableSet()
+                val variantManager = when {
+                    target.plugins.hasPlugin(AppPlugin::class.java) -> target.plugins.findPlugin(AppPlugin::class.java)?.variantManager
+                    target.plugins.hasPlugin(LibraryPlugin::class.java) -> target.plugins.findPlugin(LibraryPlugin::class.java)?.variantManager
+                    target.plugins.hasPlugin(InstantAppPlugin::class.java) -> target.plugins.findPlugin(InstantAppPlugin::class.java)?.variantManager
+                    target.plugins.hasPlugin(FeaturePlugin::class.java) -> target.plugins.findPlugin(FeaturePlugin::class.java)?.variantManager
+                    target.plugins.hasPlugin(TestPlugin::class.java) -> target.plugins.findPlugin(TestPlugin::class.java)?.variantManager
+                    else -> throw StopExecutionException("Must be applied with 'android' or 'android-library' plugin.")
```

Comment:
Are all of these types guaranteed to be on the classpath when either the `android` or the `android-library` plugin is applied?

If only the `android` plugin is applied will you get a `ClassDefNotFoundException` for the `AppPlugin` class?

Reply:
Yes, everything is ok. I've tested on my local test project.